packages/ember-glimmer/lib/utils/lookup-component.js

```diff
@@ -7,14 +8,40 @@ function lookupComponentPair(componentLookup, owner, name, options) {
   let component = componentLookup.componentFor(name, owner, options);
   let layout = componentLookup.layoutFor(name, owner, options);
 
-  if (!layout && component) {
-    let layoutProp = component.proto().layout;
-    if (layoutProp) {
-      let templateFullName = 'template:components/' + name;
-      owner.register(templateFullName, layoutProp);
+  if (component) {
+    let templateFullName = 'template:components/' + name;
+    let {
+      templateForName,
```

Comment:
@krisselden Htmlbars expects it to be on the view/component, but component init runs too late for the same logic to apply in glimmer. I was trying to keep things DRY by pulling templateForName off the class, but I can WET-ify that function if it's preferred.